database/influxdb/influxdb_retention_policy.py

```diff
@@ -0,0 +1,234 @@
+      replication: 1
+'''
+
+RETURN = '''
+#
+'''
+
+import requests.exceptions
+import re
+try:
+    from influxdb import InfluxDBClient
+    from influxdb import exceptions
+    HAS_INFLUXDB = True
+except ImportError:
+    HAS_INFLUXDB = False
+
+
+def influxdb_argument_spec():
+    return dict(
+        hostname=dict(required=True),
+        port=dict(default=8086, type='int'),
+        username=dict(default='root'),
+        password=dict(default='root', no_log=True),
+        database_name=dict(default=None)
+    )
+
+
+def connect_to_influxdb(module):
+    hostname = module.params['hostname']
+    port = module.params['port']
+    username = module.params['username']
+    password = module.params['password']
+    database_name = module.params['database_name']
+
+    client = InfluxDBClient(
+        host=hostname,
+        port=port,
+        username=username,
+        password=password,
+        database=database_name
+    )
+    return client
+
+
+def find_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+    retention_policy = None
+
+    try:
+        retention_policies = client.get_list_retention_policies(database=database_name)
+        for policy in retention_policies:
+            if policy['name'] == policy_name:
+                retention_policy = policy
+                break
+    except requests.exceptions.ConnectionError as e:
+        module.fail_json(msg=str(e))
+    return retention_policy
+
+
+def create_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+    duration = module.params['duration']
+    replication = module.params['replication']
+    default = module.params['default']
+
+    if not module.check_mode:
+        try:
+            client.create_retention_policy(policy_name, duration, replication, database_name, default)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+    module.exit_json(changed=True)
+
+
+def alter_retention_policy(module, client, retention_policy):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+    duration = module.params['duration']
+    replication = module.params['replication']
+    default = module.params['default']
+    duration_regexp = re.compile('(\d+)([hdw]{1})|(^INF$){1}')
+    changed = False
+
+    duration_lookup = duration_regexp.search(duration)
+
+    if duration_lookup.group(2) == 'h':
+        influxdb_duration_format = '%s0m0s' % duration
+    elif duration_lookup.group(2) == 'd':
+        influxdb_duration_format = '%sh0m0s' % (int(duration_lookup.group(1)) * 24)
+    elif duration_lookup.group(2) == 'w':
+        influxdb_duration_format = '%sh0m0s' % (int(duration_lookup.group(1)) * 24 * 7)
+    elif duration == 'INF':
+        influxdb_duration_format = 'INF'
+
+    if not module.check_mode:
```

Comment:
I would move this check inside the try block so the module running in check mode could report that it _would have_ made a change without taking action. Currently it would always respond with no change in check mode even if one would have occurred.